src/Blueprint/Blueprint/Cmdlets/NewAzureRMBlueprintAssignment.cs

```diff
@@ -0,0 +1,90 @@
+        public string Name { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.BlueprintObject)]
+        [ValidateNotNull]
+        public PSBlueprintBase Blueprint { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.SubscriptionIdToAssign)]
+        [ValidateNotNullOrEmpty]
+        public string[] SubscriptionId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.Location)]
+        [ValidateNotNullOrEmpty]
+        [LocationCompleter("Microsoft.Batch/operations")]
+        public string Location { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public Hashtable ResourceGroups { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.Parameters)]
+        [ValidateNotNull]
+        public Hashtable Parameters { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.CreateBlueprintAssignment, Mandatory = false, HelpMessage = ParameterHelpMessages.LockFlag)]
+        public PSLockMode? Lock { get; set; }
+        #endregion Parameters
+
+        #region Cmdlet Overrides
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+               if (ShouldProcess(Name, string.Format(Resources.CreateAssignmentShouldProcessString, Name)))
+                {
+                    var assignment = CreateAssignmentObject(ManagedServiceIdentityType.SystemAssigned,
+                                        Location,
+                                        Blueprint.Id,
+                                        Lock,
+                                        Parameters,
+                                        ResourceGroups);
+
+                    var subscriptionsList = SubscriptionId ?? new[] { DefaultContext.Subscription.Id };
+
+                    foreach (var subscription in subscriptionsList)
+                    {
+                        var scope = string.Format(BlueprintConstants.SubscriptionScope, subscription);
+                        CheckIfAssignmentExist(scope, Name, false);
+                        // First Register Blueprint RP and grant owner permission to BP service principal
+                        RegisterBlueprintRp(subscription);
+                        var servicePrincipal = GetBlueprintSpn();
+                        AssignOwnerPermission(subscription, servicePrincipal);
```

Comment:
>AssignOwnerPermission [](start = 24, length = 21)

We should probably discuss with the rest of the team, but I'm wondering if we should have a switch to make this conditional.

Reply:
I'll bring this up with the team. Adding it as an item for the next phase of cmdlets.

---
In reply to: [257388642](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257388642) [](ancestors = 257388642)